.gitlab/ci/test-native-candidate.yml

```diff
@@ -0,0 +1,69 @@
+.test-native-candidate-rule:
+  rules:
+    - if: '$CI_PIPELINE_SOURCE =~ /^(schedule|trigger)$/ && $DEMISTO_TEST_NATIVE_CANDIDATE == "true"'
+
+# used for jobs which we want to run in a pipeline even when previous jobs in the pipeline fail e.g. slack notification
+.test-native-candidate-rule-always:
+  rules:
+    - if: '$CI_PIPELINE_SOURCE =~ /^(schedule|trigger)$/ && $DEMISTO_TEST_NATIVE_CANDIDATE == "true"'
+      when: always
+
+test-native-candidate:run-lint-with-native-candidate:
+  extends:
+    - .unittests-and-lint-settings
+    - .test-native-candidate-rule
+  script:
+    - section_start "Versions"
+    - |
+      echo "demisto-sdk version: $(demisto-sdk --version)"
+      echo "mypy version: $(mypy --version)"
+      echo "flake8 py2 version: $(python2 -m flake8 --version)"
+      echo "flake8 py3 version: $(python3 -m flake8 --version)"
+      echo "bandit py2 version: $(python2 -m bandit --version 2>&1)"
+      echo "bandit py3 version: $(python3 -m bandit --version 2>&1)"
+      echo "vulture py2 version: $(python2 -m vulture --version 2>&1)"
+      echo "vulture py3 version: $(python3 -m vulture --version 2>&1)"
+    - section_end "Versions"
+    - section_start "Run Lint with Native Candidate"
+    - |
+      mkdir ./unit-tests
+      if [[ "$NATIVE_CANDIDTAE_IMAGE" == "latest" ]]; then
+        echo "----------Getting tag for latest native:dev----------"
+        NATIVE_CANDIDTAE_IMAGE=$(python3 Tests/scripts/get_latest_docker_image.py -i "demisto/py3-native")
```

Comment:
why do we need this? Isnt it the same as just running native:dev?

Reply:
This is mainly to prevent lint from getting a race condition. If the pipeline is triggered and right after that there is a newer version of native:dev, lint might test some files with the old version and some with the new. 
This is also good for just keeping track of image we test on and saving it as an artifact (and having it as a title). It will make reproducing bugs and knowing our status easier. 
If in later phases we'll want to automatically commit the name of the native:candidate, it's good that we already have it.